plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -4660,9 +4660,63 @@ public void testFileModifiedTimeHiddenColumn()
             ZonedDateTime fileModifiedTime = (ZonedDateTime) computeScalar("SELECT \"$file_modified_time\" FROM " + table.getName());
             ZonedDateTime afterTime = (ZonedDateTime) computeScalar("SELECT current_timestamp(3)");
             assertThat(fileModifiedTime).isBetween(beforeTime, afterTime);
+
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES (2)", 1);
+            ZonedDateTime anotherFileModifiedTime = (ZonedDateTime) computeScalar("SELECT max(\"$file_modified_time\") FROM " + table.getName());
+            assertNotEquals(fileModifiedTime, anotherFileModifiedTime);
```

Comment:
assert that anotherFileModifiedTime is strictly ahead of fileModifiedTime (to detect potential clock backward adjustment)

add a message

also, i think we need a `Thread.sleep(1)` to guarantee that times are indeed different.